cluster-api/pkg/controller/machineset/reconcile_test.go

```diff
@@ -196,6 +214,25 @@ func createMachineSet(replicas int, machineSetName string, machineName string, n
 }
 
 func machineFromMachineSet(machineSet *v1alpha1.MachineSet, name string) *v1alpha1.Machine {
+	amachine := machineWithoutOwnerReferences(machineSet, name)
+
+	controller := true
+	blockOwnerDeletion := true
+	amachine.ObjectMeta.OwnerReferences = []metav1.OwnerReference{
```

Comment:
why not use `*metav1.NewControllerRef(machineSet, controllerKind)`? Or you want to test the behavior of `metav1.NewControllerRef()`?

Reply:
changed to use NewControllerRef.